### ISSUE TITLE
"--exclude-resources pvc" blocks restic restore and podvolumerestore

### ISSUE THREAD

**Issue by SVronskiy** [vmware-tanzu/velero#6022]
**What steps did you take and what happened:**
Run restore, namespace was restored as expected, but volumes were not restored and  no PodVolumeRestore apppear:
```
velero restore create $backup-$(Build.SourceBranchName) --from-backup $backup --include-namespaces meanmine-prod  --restore-volumes=true  --namespace-mappings meanmine-prod:meanmine-gold --exclude-resources=pvc --wait
```

**What did you expect to happen:**
Restore backup and volume files. 

My tests show that reason of problem was in "--exclude-resources=pvc". As soon as I remove this option, volume restore run normally and PodVolumeRestore appear.

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
velero debug  --backup meanmine-prod-daily-20230323123732 --restore meanmine-prod-daily-20230323123732-gold
[bundle-2023-03-23-12-55-20.tar.gz](https://github.com/vmware-tanzu/velero/files/11051031/bundle-2023-03-23-12-55-20.tar.gz)

**Environment:**

- Velero version (use `velero version`): 
```
Client:
        Version: v1.10.2
        Git commit: 7416504e3a8fea40f78bbc4cfefc7c642aafc812
Server:
        Version: v1.10.2
```
- Velero features (use `velero client config get features`): 
```
features: <NOT SET>
```
- Kubernetes version (use `kubectl version`):
```
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.6", GitCommit:"ff2c119726cc1f8926fb0585c74b25921e866a28", GitTreeState:"clean", BuildDate:"2023-01-18T19:22:09Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.6", GitCommit:"ff2c119726cc1f8926fb0585c74b25921e866a28", GitTreeState:"clean", BuildDate:"2023-01-18T19:15:26Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Kubernetes installer & version:
```
kubeadm
```
- Cloud provider or hardware configuration:
```
Hardware server Xeon, 64 Gb Ram 1 Tb SDD
```
- OS (e.g. from `/etc/os-release`):
```
PRETTY_NAME="Ubuntu 22.04.2 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.2 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by sseago** [comment]
If you exclude PVCs from restore, then restore doesn't include PVCs. If we're not restoring PVCs, then there's no way to restore data to the (excluded) PVCs. I'm not seeing a bug here -- it's just that you need to include PVCs if you're expecting PVCs to be restored.

**Comment by SVronskiy** [comment]
Hello, @sseago !

Thank you for your answer! I'm restoring namespace for testing puposes on my PVC with cloning capability, and I do not need Velero to restore PVC, I just need Velero to restore volume data, but it doesn't happen.  I will tell you more, ci/cd with this functionanality was working for at least year and stopped working one month ago. Something changed in velero.

**Comment by blackpiglet** [comment]
From the debug bundle, I found the restore created three PodVolumeRestores and they completed successfully.

```
meanmine-prod-daily-20230321170646-gold-46vzl
meanmine-prod-daily-20230321170646-gold-9jlfn
meanmine-prod-daily-20230321170646-gold-hcrnc
```

Could you help to collect the restore that didn't create the PodVolumeRestore's debug bundle?

**Comment by SVronskiy** [comment]
Hello, @blackpiglet! 

Right you are, and thank you! There was an error in reproducing problem yesterday, I wrote 'pwc' instead of 'pvc' . I've fixed script today and run restore once again to reproduce problem and uploaded new debug file to issue description above. Can you please check it again?

**Comment by blackpiglet** [comment]
@SVronskiy 
I found a PR may related to this scenario: #4769.
This PR will skip to create PodVolumeRestore when PVC and PV are excluded.
This is used to avoid PartiallyFail when PVC and PV are not included.
I think this protection is reasonable, but the PR is already included in v1.9.0. Not sure why you just hit it recently.

**Comment by SVronskiy** [comment]
Well, not excluding PVC resolves my problem. Is it bug or feature - it's up to you.
